vars/buildPlugin.groovy

```diff
@@ -139,7 +139,29 @@ def call(Map params = [:]) {
                   mavenOptions += 'clean install'
                   try {
                     boolean usingArtifactCachingProxy = false
-                    if (artifactCachingProxyEnabled) {
+
+                    // If the build concerns a pull request, check if there is "skip-artifact-caching-proxy" label applied in case the user doesn't wan't ACP
+                    boolean prLabelsContainSkipACP = false
+                    final String skipACPLabel = 'skip-artifact-caching-proxy'
+                    if (!env.BRANCH_IS_PRIMARY) {
+                      withCredentials([
+                        // This credential only exists on ci.jenkins.io and allows to check for GitHub PR labels on @jenkinsci GitHub organisation.
+                        // Would not be needed if there was a way to retrieve pull request labels natively.
```

Comment:
@lemeurherve Why not use https://github.com/jenkinsci/pipeline-github-plugin which we already have installed on our controllers and whose `README` claims you could write

```groovy
// CHANGE_ID is set only for pull requests, so it is safe to access the pullRequest global variable
if (env.CHANGE_ID) {
  def labels = [] as Set
  pullRequest.labels.each { label ->
    labels << label
  }
  echo labels.toString() // prints the set for debugging purposes
}
```

Reply:
Because no one told me about it when I asked how I could do that, and because I haven't found out when I searched.

What would be the benefit(s) of switching to this method? (honest question)

Reply:
Note: this code has evolved since this pr, and doesn't require a credential associated with a specific instance anymore.

Reply:
Since this plugin is already an "hard" dependency of our instances I'll adapt the code to use it.
Thanks for the pointer and these explanations @basil

Reply:
Corresponding PR: https://github.com/jenkins-infra/pipeline-library/pull/635